pkg/event/target/queuestore.go

```diff
@@ -154,17 +155,28 @@ func (store *QueueStore) del(key string) error {
 	return nil
 }
 
-// ListN - lists atmost N files from the directory.
-func (store *QueueStore) ListN(n int) []string {
+// List - lists all files from the directory.
+func (store *QueueStore) List() []string {
 	store.RLock()
 	defer store.RUnlock()
-	return store.listN(n)
+	return store.list()
 }
 
 // lockless call.
-func (store *QueueStore) listN(n int) []string {
+func (store *QueueStore) list() []string {
+	var names []string
 	storeDir, _ := os.Open(store.directory)
-	names, _ := storeDir.Readdirnames(n)
+	files, _ := storeDir.Readdir(-1)
+
+	// Sort the dentries.
+	sort.SliceStable(files, func(i, j int) bool {
```

Comment:
No need to use Stable sorting @Praveenrajmani Stable sorts is a misnomer it is slower and not useful for us when our elements are not governing any pre-ordained order. 

https://github.com/minio/minio/pull/5401#discussion_r161648310

Reply:
got it :+1: